// MyApp/webpack/loaders.js

var babel_query = {
  presets: ['es2015', 'stage-2'],
  plugins: ['transform-runtime']
}

module.exports = [
  {
    test: /\.js$/,
    exclude: /(node_modules|bower_components)/,
    loader: 'babel',
    query: babel_query
  },
  {
    test: /(appril|crudle).*\.js$/,
    exclude: /(appril|crudle).*\/node_modules/,
    loader: 'babel',
    query: babel_query
  },
  { test: /\.html$/,     loader: 'vue-html' },
  { test: /\.coffee$/,   loader: 'coffee-loader' },
  { test: /\.css$/,      loader: 'style-loader!css-loader' },
  { test: /\.json$/,     loader: 'json-loader' },
  { test: /\.otf($|\?)/, loader: require.resolve('file-loader') },
  { test: /\.ttf($|\?)/, loader: require.resolve('file-loader') },
  { test: /\.eot($|\?)/, loader: require.resolve('file-loader') },
  { test: /\.svg($|\?)/, loader: require.resolve('file-loader') },
  { test: /\.png($|\?)/, loader: require.resolve('url-loader') + '?limit=10000&mimetype=image/png' },
  { test: /\.jpg($|\?)/, loader: require.resolve('file-loader') },
  { test: /\.gif($|\?)/, loader: require.resolve('file-loader') },
  { test: /\.woff[2]?($|\?)/, loader: require.resolve('url-loader') + '?limit=10000&mimetype=application/font-woff' },
]
