// MyApp/base/components/users/emailer/index.js
import MediumEditor from 'helpers/medium-editor'
import 'bootstrap/dist/js/umd/modal.js'

export default {
  props: {
    email: {
      type: String,
      required: true
    },
    send: {
      type: Function,
      required: true
    },
    close: {
      type: Function,
      required: true
    }
  },

  data() {return {
    subject: null,
    body: null
  }},

  template: require('./template'),

  mounted() {
    this.api.fetch_email(this.email).send({
      success: ({subject, body}) => {

        this.subject = subject
        this.body = body

        $(this.$el).on('shown.bs.modal', () => {

          this.editor = new MediumEditor(
            this.$el.querySelector('[name="body"]'),
            this.$el.querySelector('#toolbar')
          )

        })

        this.$nextTick(() => $(this.$el).modal({backdrop: 'static', keyboard: false}))
      }
    })
  },

  beforeDestroy() {
    $(this.$el).modal('hide')
  },

  methods: {

    trigger_send(evt) {
      let subject = this.$el.querySelector('[name="subject"]').value
      let body = this.editor.getContent()
      evt.target.disabled = true
      this.send(subject, body, () => evt.target.disabled = false)
    },

    trigger_close() {
      this.close()
    }
  }
}
