// MyApp/base/components/orders/editor/create/cart/methods.js
import {map} from 'lodash'
import Alert from 'appril-alert'
import {product_photo} from 'components/orders/helpers'

export function photo(index) {
  let row = this.cart[index]
  return product_photo.call(this, row.product, (row.sku || {}).color)
}

export function plus(index) {
  this.$store.commit('ORDERS__CART__PLUS', index)
}

export function minus(index) {
  if (this.cart[index].qty > 1)
    this.$store.commit('ORDERS__CART__MINUS', index)
}

export function remove(index) {
  this.$store.commit('ORDERS__CART__REMOVE', index)
}

export function place_order() {
  if (!this.user)
    return Alert.warning('Select user please')

  let cart = map(JSON.parse(JSON.stringify(this.cart)), (row) => {
    row.product = row.product.id
    if (row.skus) {
      row.skus = map(row.skus, 'id')
    } else {
      row.sku = row.sku.id
    }
    return row
  })

  this.api.place_order(this.url.params.id, this.user.id, cart).send({
    success: (order) => {
      this.$store.commit('ORDERS__UPDATE', order)
      this.$store.commit('ORDERS__RESET')
    }
  })
}
