// MyApp/base/components/orders/helpers.js
import {find} from 'lodash'
import Vue from 'vue/dist/vue.js'
import {generate_uuid} from 'appril-utils'

import receipt from './receipt'
import print_styles from 'base_templates/print_styles'

export const pixel = 'data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACwAAAAAAQABAAACAkQBADs='

let print_iframe
const print_iframe_name = `print_iframe-${generate_uuid()}`
let iframe = document.createElement('iframe')
iframe.style.display = 'none'
iframe.setAttribute('name', print_iframe_name)
iframe.src = 'about:blank'
iframe.onload = () => print_iframe = window.frames[print_iframe_name]
document.body.appendChild(iframe)

export function print_orders(orders) {

  this.api.selected(orders).send({
    success: (orders) => {

      let vm = new Vue({
        el: document.createElement('div'),
        data: {orders},
        template: '<div><receipt v-for="order in orders" :order="order" /></div>',
        components: {receipt}
      })

      let html = vm.$el.innerHTML
      vm.$el.remove()
      vm.$destroy()

      print_iframe.document.body.innerHTML = html + print_styles
      print_iframe.window.focus()
      print_iframe.window.print()
    }
  })
}

export function product_photo(product, color) {
  let photo = (product.color_photos || {})[color]

  if (!photo) {
    if (!product.photos)
      return pixel

    if (!product.photos[0])
      return pixel

    if (!product.photos[0].thumb)
      return pixel

    photo = product.photos[0]
  }

  return [
    this.env.product_photos_url,
    photo.thumb
  ].join('/')
}

export function product_default_sku(product) {
  return find(product.skus, (s) => s.default)
}
