// MyApp/base/components/orders/editor/receipt/methods.js
import Alert from 'appril-alert'
import {print_orders} from '../../helpers'

export function activate_shipment_controls() {
  this.$store.commit('ORDERS__TOGGLE_SHIPMENT_CONTROLS', true)
}

export function close_shipment_controls() {
  this.$store.commit('ORDERS__TOGGLE_SHIPMENT_CONTROLS', false)
}


export function update_tracking_number() {

  let tracking_number = this.$el.querySelector('input[name="tracking_number"]').value

  if (!tracking_number)
    return Alert.error('Please insert tracking number')

  let carrier = this.$el.querySelector('input[name="carrier"]').value

  if (!carrier)
    return Alert.error('Please insert carrier')

  this.api.update_tracking_number(this.order.id, tracking_number, carrier).send({
    success: () => {
      this.$store.commit('ORDERS__UPDATE_TRACKING_NUMBER', {tracking_number, carrier})
      this.close_shipment_controls()
    }
  })

}


export function print() {
  print_orders.call(this, [this.order.id])
}


export function update_status(status) {

  Alert.confirm(`You are about to ${status} this order`, () => {
    this.api.update_status(this.order.id, status).send({
      success: (order) => this.$store.commit('ORDERS__UPDATE', order)
    })
  })
}
