// MyApp/base/components/models/editor/credits/methods.js
import Serialize from 'form-serialize'
import MD5 from 'blueimp-md5'
import Alert from 'appril-alert'

function with_password_prompt(callback) {
  Alert({
    title: 'Password Please',
    type: 'input',
    inputType: 'password',
    showCancelButton: true
  }, callback)
}

export function create(evt) {

  let form = evt.target.closest('form')

  if (!$(form).parsley().validate())
    return

  with_password_prompt((password) => {
    if (!password)
      return

    let data = Serialize(form, {hash: true})

    this.api.add_credits(MD5(password), this.item.id, data).send({
      success: (credits) => {
        this.$store.commit('USERS__SET_CREDITS', credits)
        form.reset()
      }
    })
  })
}

export function remove(credit_id, evt) {

  with_password_prompt((password) => {
    if (!password)
      return

    this.api.delete_credits(MD5(password), this.url.params.id, credit_id).send({
      success: (credits) => {
        this.$store.commit('USERS__SET_CREDITS', credits)
      }
    })
  })

}
