// MyApp/base/components/models/editor/profile/status/index.js
import {omit} from 'lodash'
import Alert from 'appril-alert'

import member_status_map from '../../../member_status_map'
import member_type_map from '../../../member_type_map'

import Emailer from 'components/users/emailer'

export default {
  props: ['item'],
  template: require('./template'),

  data: () => ({
    member_status_map,
    member_type_map,
    email: null
  }),

  components: {
    Emailer
  },

  computed: {
    member_statuses() {
      return omit(member_status_map, 'approved')
    },
  },

  methods: {

    update_status(status) {

      if (status === this.$options.rejected_status) {

        this.email = 'model_status_changed_to_rejected'

      } else {
        Alert.confirm(`You are about to update status to ${status}`, () => {
          this.api.update_member_status(this.item.id, status).send({
            success: (updates) => this.$store.commit('CRUDLE__UPDATE_ITEM', updates)
          })
        })
      }
    },

    reject(subject, body) {

      this.api.reject(this.item.id, subject, body).send({
        success: (updates) => {
          this.close()
          this.$store.commit('CRUDLE__UPDATE_ITEM', updates)
        }
      })
    },

    close() {
      this.email = null
    }
  }
}
